modules.d/95iscsi/iscsiroot.sh

```diff
@@ -229,7 +229,7 @@ handle_netroot() {
             echo "$target"
         done
     })
-    [ -z "$targets" ] && warn "Target discovery to $iscsi_target_ip:${iscsi_target_port:+$iscsi_target_port} failed with status $?" && return 1
+    [ -z "$targets" ] && warn "Target discovery to $iscsi_target_ip:${iscsi_target_port:+$iscsi_target_port} failed" && return 1
```

Comment:
I think that `&&` pipelines should be avoided, in particular for tests like this.
I suggest

```
[ ! -z "$targets" ] || { warn "blah"; return 1; }
```
or using an `if` clause.

Reply:
Why should && pipelines be avoided?

An `if` clause would probably the most readable code. Patches are welcome. :)

Reply:
> Why should && pipelines be avoided?

Because the pipeline's exit status is *failed* if any command in the pipeline fails. That may not hurt here, but it definitely does if you use `errexit`.

For example, people write code like this:

```bash
[ -f some_file ] && rm -f some_file
```
With `errexit` set, this will cause your script to abort if the file doesn't exist, which is probably not what you intended.
```bash
[ ! -f some_file ] || rm -f some_file
```
works though, and has exactly the same semantics as
```bash
if [ -f some_file ]; then rm -f some_file; fi
```
Which one is more readable is a matter of taste.

I for one use `errexit` a lot, and therefore I got used to avoiding `&&` pipelines completely.